### ISSUE TITLE
noUnusedLocals flags decorated classes

### ISSUE THREAD

**Issue by roblourens** [Microsoft/TypeScript#13853]
**TypeScript Version:**  2.1.5 and latest 2.2.0

**Code**

```ts
@f
class A {
    private b;

    c() {

    }
}

export function f(ctor) {
    new ctor().c();
}
```

with tsconfig.json:
```json
{
    "compilerOptions": {
        "noUnusedLocals": true,
        "target": "es6",
        "module": "commonjs",
        "experimentalDecorators": true
    }
}
```

**Expected behavior:**
`A.b` should be flagged as unused, but `A` should not be flagged as unused, because the decorator could be doing something useful with the class. This is a common pattern in our codebase.

**Actual behavior:**
`A` is flagged as unused

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/13120

![image](https://cloud.githubusercontent.com/assets/8000722/22571494/c06163ae-e954-11e6-8b2a-8152ea8afa7a.png)
